### ISSUE TITLE
 WhatsApp error: unexpected websocket close: websocket: close 1006 (abnormal closure): unexpected EOF

### ISSUE THREAD

**Issue by gzlivre** [tulir/mautrix-whatsapp#35]
Hi,
After a few minutes of starting the service mautrix-whatsapp, I believe that whatsapp is blocking my requests.
this is de log:
```
error writing to socket: write tcp [XXXX:xXXXX:XXXXX:XXXX:XXX]:53642->[2a03:2880:f204:c5:face:b00c:0:167]:443: write: connection reset by peer
[Feb 22, 2019 17:06:46] [User/@user:matrix.domain.org/ERROR] WhatsApp error: unexpected websocket close: websocket: close 1006 (abnormal closure): unexpected EOF
[Feb 22, 2019 17:06:47] [User/@user:matrix.domain.org/ERROR] WhatsApp error: unexpected websocket close: websocket: close 1006 (abnormal closure): unexpected EOF
[Feb 22, 2019 17:06:48] [User/@user:matrix.domain.org/ERROR] WhatsApp error: unexpected websocket close: websocket: close 1006 (abnormal closure): unexpected EOF
[Feb 22, 2019 17:06:49] [User/@user:matrix.domain.org/ERROR] WhatsApp error: unexpected websocket close: websocket: close 1006 (abnormal closure): unexpected EOF
[Feb 22, 2019 17:06:49] [User/@user:matrix.domain.org/ERROR] WhatsApp error: unexpected websocket close: websocket: close 1006 (abnormal closure): unexpected EOF
```
# OS version
```
# cat /etc/issue
Debian GNU/Linux 9 \n \l
```

# go version
```
:~# go version
go version go1.11.5 linux/amd64
```

# mautrix-whatsapp version
```
:~/mautrix-whatsapp# git pull
Already up-to-date.
```

**Comment by vorletzter** [comment]
Its a bug in a Library: https://github.com/tulir/mautrix-whatsapp/issues/35

**Comment by babolivier** [comment]
Also a duplicate of #21

**Comment by tulir** [comment]
#21 is for normal closures (e.g. opening whatsapp web), this is a different issue that was probably temporarily fixed by downgrading go-whatsapp

**Comment by babolivier** [comment]
Oh, my bad then.

**Comment by gzlivre** [comment]
thanks to everyone
I will wait for the go-whatsapp update

**Comment by gzlivre** [comment]
i pull the latests commits, and now its works perfect
maybe https://github.com/tulir/mautrix-whatsapp/commit/a5f5d93d9f21077e3801cee92a6e49239f78e5a6

**Comment by tulir** [comment]
@gzlivre As I said, downgrading go-whatsapp fixed it temporarily, so 6dd83b3d30d68dc1626dca1b43ce93e3a6f1e672 and 485acf6de654b8fb70007876c074fb004eb9717b mostly

**Comment by gzlivre** [comment]
@tulir, many thanks
great job!
